### PR TITLE
Import by local authority

### PR DESCRIPTION
**Pull request by notapatch**
Importing by local authority

### CODE REVIEW

File: app/importers/planning_applications_importer.rb [hunk 1]
```diff
@@ -48,12 +52,8 @@ def import_row(row)
      end
    end
 
-   # Not sure how to get the local_authority
-   # Suggestion was from the file name
-   # I would need to know that this was the file to import.
-   # File name could have a date and I could keep a "last imported local_authority: Buckinghamshire, date: 3/9/2023"
    def local_authority
-     @local_authority ||= LocalAuthority.find_or_create_by(name: "Bucks")
+     @local_authority ||= LocalAuthority.find_by!(name: @local_authority_name)
```
Comment (allbecauseyoutoldmeso) [inline app/importers/planning_applications_importer.rb:56]: possibly overkill but I'd be tempted to add a private `attr_reader :local_authority_name` and then make this `name: local_authority_name`

File: spec/importers/planning_applications_importer/address_spec.rb [hunk 2]
```diff
@@ -99,7 +102,11 @@
...
         .with(%[Planning application reference: 22/06867/PNP6A has invalid property: Full can't be blank].strip)
 
-      PlanningApplicationsImporter.new.call
+      importer
     end
   end
+
+  def importer
+    PlanningApplicationsImporter.new(local_authority_name: "lambeth").call
+  end
```
Comment (allbecauseyoutoldmeso) [inline spec/importers/planning_applications_importer/address_spec.rb:111]: could you have `let(:importer) { PlanningApplicationsImporter.new(local_authority_name: "lambeth").call }`?

File: spec/importers/planning_applications_importer/plannint_application_attributes_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
+require "rails_helper"
```
Comment (allbecauseyoutoldmeso) [inline spec/importers/planning_applications_importer/plannint_application_attributes_spec.rb:1]: file name typo 😆 'plannin**t**_application_attributes_spec.rb'

**Review by allbecauseyoutoldmeso — APPROVED** [review]
🙌

File: spec/importers/planning_applications_importer/plannint_application_attributes_spec.rb (hunk 3 above)
Comment (notapatch) [reply spec/importers/planning_applications_importer/plannint_application_attributes_spec.rb:1]: :nod:

File: spec/importers/planning_applications_importer/address_spec.rb (hunk 2 above)
Comment (notapatch) [reply spec/importers/planning_applications_importer/address_spec.rb:111]: It should be noted that my "let" fu is weak so thank you.

I'll make the suggested change across the whole PR except the planning_application_importer_spec.rb which has an argument to the test importer function.  To get around that I would have to create a extra nested block to change the local_aurthority_name for the one test :eyeroll:

File: app/importers/planning_applications_importer.rb (hunk 1 above)
Comment (notapatch) [reply app/importers/planning_applications_importer.rb]: Bit more verbose. A bit more consistent usage of access. I'll try it. :nod:

**Pull request merged**